src/deroff.rs

```diff
@@ -556,6 +557,23 @@ impl Deroffer {
         unimplemented!()
     }
 
+    /// `condputs` (cond)itionally (puts) `s` into `self.output`
+    /// if `self.tr` is set, instead of putting `s` into `self.output` directly,
+    /// it `translate`s it using the set translation table and puts the result
+    /// into `self.output`
+    fn condputs(&mut self, s: &str) {
+        let is_special =
+            { self.pic || self.eqn || self.refer || self.r#macro || self.inlist || self.inheader };
+
+        if !is_special {
+            if let Some(table) = &self.tr {
+                self.output.push_str(&table.translate(s.into()));
```

Comment:
Can get rid of the conversion into a String if we allow `translate` to take a `&str`.
```suggestion
                self.output.push_str(&table.translate(s));
```